src/lib/api/api.go

```diff
@@ -196,25 +198,54 @@ func (a *API) Query(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
+	if hash == "" {
+		switch Type {
+		case "system":
+			timeSeriesPath = "storage/system" + tsdb.FileExtension
+		case "journal":
+			timeSeriesPath = "storage/journal" + tsdb.FileExtension
+		default:
+			a.send(w, []byte("INVALID_TYPE"))
+			return
+		}
+		if ok := tsdb.VerifyChainPathExists(timeSeriesPath); !ok {
+			a.send(w, []byte("INVALID_PATH"))
+			return
+		}
+		newBlocks = []tsdb.Block{}
+	} else {
+		if _, ok := (*a.Matrix)[hash]; !ok {
+			a.send(w, []byte("INVALID_PATH"))
+			return
+		}
+		matrix := (*a.Matrix)[hash]
+
+		switch Type {
+		case "ping":
+			newBlocks = matrix.PingChain.Chain
+		case "jitter":
+			newBlocks = matrix.JitterChain.Chain
+		default:
+			a.send(w, []byte("INVALID_TYPE"))
+			return
+		}
+
+		timeSeriesPath = "storage/" + Type + "/chunk_" + Type + "_" + hash + tsdb.FileExtension
+	}
 	// condition: only for bench-routes as per the design
 	//
 	// path should be in syntax: <DBPath>/<ofType>/chunk_<middle>_<url>.json -> non-system metric
 	// %s/system.json -> system metric
 
 	// verify if chain path exists
-	timeSeriesPath = timeSeriesPath + tsdb.FileExtension
-	if ok := tsdb.VerifyChainPathExists(timeSeriesPath); !ok {
-		a.send(w, []byte("INVALID_PATH"))
-		return
-	}
 
 	// TODO: we do not capture the block streams in memory while querying yet. They are captured only when flushed.
 	// TODO: consider cmap while querying for fetching latest blocks after shifting tsdb to binary.
 
 	qry := querier.New(timeSeriesPath, "", querier.TypeRange)
 	query := qry.QueryBuilder()
 	query.SetRange(startTimestamp, endTimestamp)
-	a.Data = query.ExecWithoutEncode()
+	a.Data = query.ExecWithoutEncode(newBlocks)
```

Comment:
I see newBlocks is always going to be empty? So is there a reason to pass this down the func?

Reply:
No not-neccessary as : 
```
		case "ping":
			newBlocks = matrix.PingChain.Chain
		case "jitter":
			newBlocks = matrix.JitterChain.Chain
		default:
```
`newBlocks` will have the unflushed memory-chains which are monitored and may have new blocks.

Reply:
I am having this inference from main.go : 
```
		chainSet.Register(fmt.Sprintf("%s-ping", uHash), targetMachineCalc[uHash].Ping)
		chainSet.Register(fmt.Sprintf("%s-jitter", uHash), targetMachineCalc[uHash].Jitter)
		chainSet.Register(fmt.Sprintf("%s-fping", uHash), targetMachineCalc[uHash].FPing)
```
as targetMachineCalc[uHash] are registered in the `chainset` and also,
```
	        matrix[hash] = &utils.BRMatrix{
		        Route:        r,
		        PingChain:    targetMachineCalc[uHash].Ping,
		        JitterChain:  targetMachineCalc[uHash].Jitter,
		        FPingChain:   targetMachineCalc[uHash].FPing,
		        MonitorChain: tsdb.NewChain(pathMonitor).Init(),
		        Metrics:      endpointMetrics,
	        }
```
targetMachineCalc[uHash] is assigned to the matrix , so when : 
```
		        case "Ping":
			        workers.Ping.Iterate("stop", false)
		        case "FloodPing":
			        workers.PingF.Iteratef("stop", false)
		        case "Jitter":
			        workers.Jitter.Iterate("stop", false)
		        case "ReqResDelayMonitoring":
			        workers.Monitor.Iterate("stop", false)
		        }
```
It is used to iterate over all the routes the through `targetMachineCalc[uHash]`, in-memory chains will be updated in matrix as in worker we pass `targetMachineCalc[uHash]`,
```
	workers := &struct {
		Ping    *ping.Ping
		Jitter  *jitter.Jitter
		PingF   *ping.FloodPing
		Monitor *monitor.Monitor
	}{
		Ping:    ping.New(conf, ping.TestInterval{OfType: intervals[0].Type, Duration: *intervals[0].Duration}, &targetMachineCalc),
		Jitter:  jitter.New(conf, jitter.TestInterval{OfType: intervals[0].Type, Duration: *intervals[1].Duration}, &targetMachineCalc),
		PingF:   ping.Newf(conf, ping.TestInterval{OfType: intervals[0].Type, Duration: *intervals[0].Duration}, conf.Config.Password, &targetMachineCalc),
		Monitor: monitor.New(conf, monitor.TestInterval{OfType: intervals[2].Type, Duration: *intervals[2].Duration}, &matrix),
	}
```